sdk/tracing/azotel/otel.go

```diff
@@ -0,0 +1,120 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// azotel adapts OpenTelemetry tracing for consumption by the azcore/tracing package.
+package azotel
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/tracing"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	tracesdk "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// TracingProviderOptions contains the optional values for NewTracingProvider.
+type TracingProviderOptions struct {
+	// for future expansion
+}
+
+// NewTracingProvider creates a new tracing.Provider that wraps the specified OpenTelemetry TracerProvider.
+//   - tracerProvider - the TracerProvider to wrap
+//   - opts - optional configuration. pass nil to accept the default values
+func NewTracingProvider(tracerProvider *tracesdk.TracerProvider, opts *TracingProviderOptions) tracing.Provider {
+	return tracing.NewProvider(func(namespace, version string) tracing.Tracer {
+		tracer := tracerProvider.Tracer(namespace, trace.WithInstrumentationVersion(version))
+
+		return tracing.NewTracer(func(ctx context.Context, spanName string, options *tracing.SpanOptions) (context.Context, tracing.Span) {
+			kind := tracing.SpanKindInternal
+			var attrs []attribute.KeyValue
+			if options != nil {
+				kind = options.Kind
+				attrs = convertAttributes(options.Attributes)
+			}
+			ctx, span := tracer.Start(ctx, spanName, trace.WithSpanKind(convertSpanKind(kind)), trace.WithAttributes(attrs...))
+			return ctx, convertSpan(span)
+		}, &tracing.TracerOptions{
+			SpanFromContext: func(ctx context.Context) (tracing.Span, bool) {
+				return convertSpan(trace.SpanFromContext(ctx)), true
+			},
+		})
+
+	}, nil)
+}
+
+func convertSpan(traceSpan trace.Span) tracing.Span {
+	impl := tracing.SpanImpl{
+		End: func() {
```

Comment:
NIT: Signature matches for `End` and `traceSpan.End`?

Reply:
@lmolkova do we want optional values for `End()`?  I don't recall discussing this.

Reply:
only as an extensibility point in case we want to set status along with end or if OTel decides do add another end method with parameters - there were some talks about it.

Reply:
Is it enough to call `SetStatus()` before calling `End()` (that's how we currently do it in `azcore`).